src/components/Stack/Stack.tsx

```diff
@@ -28,6 +28,8 @@ export interface Props {
   alignment?: Alignment;
   /** Adjust horizontal alignment of elements */
   distribution?: Distribution;
+  /** Fill the available horizontal space in the stack with the item  */
+  fill?: boolean;
```

Comment:
I think this may be in the wrong place. The top-level stack component uses a distribution prop with `fill` as a type.

Have a look at the props for the item component. It has a fill prop already: https://github.com/Shopify/polaris-react/blob/master/src/components/Stack/components/Item/Item.tsx#L7